File: perl/2015/day01a/day01a.pl
#!/usr/bin/env perl

use strict;
use warnings;

sub usage {
    my $progname = $0;
    print "usage: $progname <input file>\n";
    exit 1;
}

sub process {
    my ($filename) = @_;
    my $counter = 0;

    open ( my $infile, '<', $filename ) or die "cannot open file: $filename: $!";
    while ( my $line = <$infile> ) {
        foreach my $ch ( split //, $line ) {
            if ( $ch eq '(' ) {
                $counter++;
            } elsif ( $ch eq ')' ) {
                $counter--;
            }
        }
    }
    close($infile);

    return $counter;
}

sub main {
    if ( @ARGV < 1 ) {
        usage();
    }
    my $filename = $ARGV[0];
    my $result = process($filename);
    print "result = $result\n";
}

main();

File: ruby/2015/day01a/day01a.rb
#!/usr/bin/env ruby

def usage
    progname = File.basename($0)
    puts "usage: #{progname} <input file>"
    exit 1
end

def process(filename)
    counter = 0
    File.foreach(filename) do |line|
        line.each_char do |ch|
            if ch == '('
                counter += 1
            elsif ch == ')'
                counter -= 1
            end
        end
    end
    counter
end

def main
    usage if ARGV.length < 1
    filename = ARGV[0]
    result = process(filename)
    puts "result = #{result}"
end

main if __FILE__ == $0
